base_layer/core/src/consensus/consensus_encoding/hashing.rs

```diff
@@ -41,8 +42,9 @@ where D: Default
 {
     #[allow(clippy::new_ret_no_self)]
     pub fn new(label: &'static str) -> ConsensusHasher<D> {
+        let network = *CURRENT_NETWORK.lock().unwrap();
```

Comment:
Would there be any critical consequences if this were to [panic](https://doc.rust-lang.org/std/sync/struct.Mutex.html#panics)?

Reply:
I think it would result in an unusable service. Which is actually a pretty good scenario. I'd rather the service crash than do something like mangle or produce bad transactions.